cylc/flow/ws_data_mgr.py

```diff
@@ -361,22 +419,41 @@ def update_task_proxies(self, task_ids=None):
                 prereq_obj = prereq.api_dump(self.workflow_id)
                 if prereq_obj:
                     prereq_list.append(prereq_obj)
+            del tproxy.prerequisites[:]
             tproxy.prerequisites.extend(prereq_list)
-            for _, msg, is_completed in itask.state.outputs.get_all():
-                tproxy.outputs.append(f"{msg}={is_completed}")
+            tproxy.outputs[:] = [
+                f'{msg}={is_completed}'
+                for _, msg, is_completed in itask.state.outputs.get_all()
+            ]
+            # Replace the original
+            # (atomic update, for access from other threads).
+            self.task_proxies[itask.identity] = tproxy
+
+        # Recalculate effected task def elements elapsed time.
+        for name, tdef in task_defs.items():
+            elapsed_time = task_mean_elapsed_time(tdef)
+            if elapsed_time:
+                task = self.tasks[name]
+                task.checksum = f'{name}@{update_time}'
```

Comment:
Checksum?

Reply:
Checksum is for future use with UIS data store sync  (it's essentially just a timestamped ID)..

Reply:
But it's not actually a "checksum"?  https://www.howtogeek.com/363735/what-is-a-checksum-and-why-should-you-care/

Reply:
Perhaps we can just call this variable a `label`, a `stamp` or something else?

Reply:
Yes, not actually a checksum.. But they are meant to make up the skeleton of the data store, which would be like a unique picture of it in a particular state (so any little change in the state would result in a different skeleton).

(so hashing the skeleton would kinda be like a checksum)

Reply:
I think we're just complaining about calling it a checksum when it's not a checksum - it may not be obvious elsewhere in the code, so is misleading.

Reply:
`stamp` then?